modules/homa-lite/src/lib.rs

```diff
@@ -1008,6 +1042,54 @@ pub mod module {
 				Ok(())
 			})
 		}
+
+		/// Helper function that iterates `RedeemRequests` storage from `NextRedeemRequestToMatch`.
+		/// If `NextRedeemRequestToMatch` is not found in storage, iterate from the start.
+		pub fn iterate_from_next_redeem_request(
+			f: &mut impl FnMut(T::AccountId, Balance, Permill) -> Result<bool, DispatchError>,
+		) -> DispatchResult {
+			let first_element = match RedeemRequests::<T>::iter().next() {
+				// Current storage is empty. Do nothing and return
+				None => return Ok(()),
+				Some(element) => element,
+			};
+
+			let starting_element = match Self::redeem_requests(Self::next_redeem_request_to_match()) {
+				Some(request) => (Self::next_redeem_request_to_match(), request),
+				None => first_element,
+			};
+			let mut iterator = RedeemRequests::<T>::iter();
+			let mut current = iterator.next();
+
+			// Skip elements until `starting_element` is found.
+			while current.is_some() && current.clone().unwrap_or_default() != starting_element {
```

Comment:
No this is very inefficient.

Do something like this

https://github.com/AcalaNetwork/Acala/blob/7f67fb914b19347eccd03d042daaa4b902b816e3/modules/cdp-engine/src/lib.rs#L664-L669

Reply:
Replaced the first iter with iter_from instead